src/system/game-data.ts

```diff
@@ -795,44 +791,49 @@ export class GameData {
     return true; // TODO: is `true` the correct return value?
   }
 
-  public saveTutorialFlag(tutorial: Tutorial, flag: boolean): boolean {
+  /**
+   * Retrieve the tutorial seen flags from local storage
+   * @returns the flags saved in local storage if they exist, otherwise an empty array
+   */
+  private getTutorialFlags(): (boolean | null)[] {
     const key = getDataTypeKey(GameDataType.TUTORIALS);
-    let tutorials: object = {};
-    if (localStorage.hasOwnProperty(key)) {
-      tutorials = JSON.parse(localStorage.getItem(key)!); // TODO: is this bang correct?
-    }
+    let tutorials = [];
 
-    Object.keys(Tutorial).forEach((key) => {
-      if (key === Tutorial[tutorial]) {
-        tutorials[key] = flag;
-      } else {
-        tutorials[key] ??= false;
+    if (localStorage.hasOwnProperty(key)) {
+      try {
+        tutorials = JSON.parse(localStorage.getItem(key)!);
+      } catch (err) {
+        console.warn("Failed to parse tutorial data from local storage", err);
```

Comment:
Please don't use `hasOwnProperty`. The ts compiler can't make the connection that way.
Instead simply use `getItem` and check the content:

```suggestion
    const lsItem = localStorage.getItem(key);
    if (lsItem) {
      try {
        tutorials = JSON.parse(lsItem);
      } catch (err) {
        console.warn("Failed to parse tutorial data from local storage", err);
```